pkg/sql/opt_catalog.go

```diff
@@ -513,6 +517,35 @@ func (oi *optIndex) Column(i int) opt.IndexColumn {
 	return opt.IndexColumn{Column: oi.tab.Column(ord), Ordinal: ord}
 }
 
+// ForeignKey is part of the opt.Index interface.
+func (oi *optIndex) ForeignKey() opt.Index {
+	if !oi.desc.ForeignKey.IsSet() {
+		return nil
+	}
+
+	indexID := oi.desc.ForeignKey.Index
+	index, err := oi.tab.cat.ResolveDataSourceByID(context.TODO(), int64(indexID))
```

Comment:
Particularly unsure about whether this is the right strategy to get to an index reference when necessary. Open to suggestions on how to make this fit best with the usual ways of generating catalog structs.